### PR TITLE
[core] Introduce expire_snapshots procedure

### PR DESCRIPTION
**Pull request by JingsongLi**

### CODE REVIEW

File: docs/content/engines/flink.md [hunk 1]
```diff
@@ -502,5 +502,18 @@ table options syntax: we use string to represent table options. The format is 'k
...
+   <tr>
+      <td>expire_snapshots</td>
+      <td>
+         -- expires snapshot<br/>
+         CALL sys.expire_snapshots('identifier', retainMax)<br/><br/>
+      </td>
+      <td>
+         To expire snapshots. Argument:
+            <li>identifier: the target table identifier. Cannot be empty.</li>
+            <li>retainMax: the maximum number of completed snapshots to retain.</li>
```
[the remark this answers is not in the document]
Comment (JingsongLi) [reply docs/content/engines/flink.md:514]: Now, Flink SQL dose not have named arguement, so here, we just provide retainMax only.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/ExpireSnapshotsProcedure.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public String identifier() {
+        return "expire_snapshots";
+    }
+
+    public String[] call(ProcedureContext procedureContext, String tableId, int retainMax)
+            throws Catalog.TableNotExistException {
+        return new String[] {
+            table(tableId).newExpireSnapshots().retainMax(retainMax).expire() + ""
```
[the remark this answers is not in the document]
Comment (JingsongLi) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/ExpireSnapshotsProcedure.java:36]: `String.valueOf`? Seems same to me.

File: docs/content/engines/flink.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (TaoZex) [reply docs/content/engines/flink.md:514]: Thanks for your reply. I understand it.

**Review by yuzelin — APPROVED** [review]
+1

**Pull request merged**